src/mlpack/tests/feedforward_network_test.cpp

```diff
@@ -621,4 +623,86 @@ BOOST_AUTO_TEST_CASE(OptimizerTest)
   model.Train(trainData, trainLabels, opt);
 }
 
+/**
+ * Train the RBF network on a larger dataset.
+ */
+BOOST_AUTO_TEST_CASE(RBFNetworkTest)
+{
+  // Load the dataset.
+  arma::mat trainData;
+  data::Load("thyroid_train.csv", trainData, true);
+
+  arma::mat trainLabels = trainData.row(trainData.n_rows - 1);
+  trainData.shed_row(trainData.n_rows - 1);
+
+  arma::mat trainLabels1 = arma::zeros(3, trainData.n_cols);
+  for(size_t i = 0; i < trainData.n_cols; i++)
+  {
+    trainLabels1.col(i).row((trainLabels(i) - 1)) = 1;
+  }
+
+  arma::mat testData;
+  data::Load("thyroid_test.csv", testData, true);
+
+  arma::mat testLabels = testData.row(testData.n_rows - 1);
+  testData.shed_row(testData.n_rows - 1);
+
+  /*
+   * Construct a feed forward network with trainData.n_rows input nodes,
+   * hiddenLayerSize hidden nodes and trainLabels.n_rows output nodes. The
+   * network structure looks like:
+   *
+   *  Input         RBF          Activation    Output
+   *  Layer         Layer         Layer        Layer
+   * +-----+       +-----+       +-----+       +-----+
+   * |     |       |     |       |     |       |     |
+   * |     +------>|     +------>|     +------>|     |
+   * |     |       |     |       |     |       |     |
+   * +-----+       +--+--+       +-----+       +-----+
+   */
+
+  arma::mat centroids;
+  KMeans<> kmeans;
+  kmeans.Cluster(trainData, 8, centroids);
+
+  FFN<MeanSquaredError<> > model;
+  model.Add<RBF<> >(trainData.n_rows, 8, centroids);
+  model.Add<Linear<> >(8, 3);
+
+  // RBFN neural net with MeanSquaredError.
+  TestNetwork<>(model, trainData, trainLabels1, testData, testLabels, 10, 0.1);
+
+  arma::mat dataset;
+  dataset.load("mnist_first250_training_4s_and_9s.arm");
+
+  // Normalize each point since these are images.
+  for (size_t i = 0; i < dataset.n_cols; ++i)
+  {
+    dataset.col(i) /= norm(dataset.col(i), 2);
+  }
+
+  arma::mat labels = arma::zeros(1, dataset.n_cols);
+  labels.submat(0, labels.n_cols / 2, 0, labels.n_cols - 1).fill(1);
+
+  arma::mat labels1 = arma::zeros(2, dataset.n_cols);
+  for(size_t i = 0; i < dataset.n_cols; i++)
+  {
+    labels1.col(i).row(labels(i)) = 1;
+  }
+  labels += 1;
+
+
+  arma::mat centroids1;
+  arma::Row<size_t> assignments;
+  KMeans<> kmeans1;
+  kmeans1.Cluster(dataset, 60, centroids1);
+
+  FFN<MeanSquaredError<> > model1;
+  model1.Add<RBF<> >(dataset.n_rows, 60, centroids1);
+  model1.Add<Linear<> >(60, 2);
+
+  // RBFN neural net with MeanSquaredError.
+  TestNetwork<>(model1, dataset, labels1, dataset, labels, 10, 0.2);
```

Comment:
Can you lower the bound for the error to 0.1 since we were able to achieve 0.03?

Reply:
can you try testing with random seed and report the results on the PR?
https://github.com/mlpack/mlpack/wiki/Testing-Guidelines#tests-with-random-input

Reply:
you should be able to run the tests locally.

Reply:
Previously it was passing locally but I updated the PR some online builds failed.

Reply:
Hey can you run the test locally with random seed and see the classification error it reports and then adjust the beta and number of centres accordingly to achieve the lowest error?

Reply:
I have tested with random seed locally.
When without setting beta parameter
```
classificationError = 0.174   for centroids = 10
classificationError = 0.13   for centroids = 20
classificationError = 0.092   for centroids = 30
classificationError = 0.092   for centroids = 40
classificationError = 0.092   for centroids = 50
classificationError = 0.078   for centroids = 60
classificationError = 0.068   for centroids = 70
classificationError = 0.088   for centroids = 80
classificationError = 0.072   for centroids = 90
classificationError = 0.07   for centroids = 100
classificationError = 0.074   for centroids = 110
classificationError = 0.09   for centroids = 120
classificationError = 0.084   for centroids = 130
classificationError = 0.06   for centroids = 140
classificationError = 0.094   for centroids = 150
```
When setting `betas=4.1`
```
classificationError = 0.192   for centroids = 10
classificationError = 0.124   for centroids = 20
classificationError = 0.07   for centroids = 30
classificationError = 0.082   for centroids = 40
classificationError = 0.032   for centroids = 50
classificationError = 0.068   for centroids = 60
classificationError = 0.038   for centroids = 70
classificationError = 0.042   for centroids = 80
classificationError = 0.046   for centroids = 90
classificationError = 0.02   for centroids = 100
classificationError = 0.03   for centroids = 110
classificationError = 0.026   for centroids = 120
classificationError = 0.032   for centroids = 130
classificationError = 0.02   for centroids = 140
classificationError = 0.026   for centroids = 150
```